src/clj/game/cards/resources.clj

```diff
@@ -1928,22 +1925,23 @@
              (:x-cost eid))
          (= :play (:source-type eid))))
     ;; companion-builder: turn-ends-ability
-    {:prompt "Trash 1 card from your grip at random or trash Mystic Maemi?"
-     :choices (req (if (can-pay? state :runner
+    {:prompt "Choose one"
+     :choices (req [(when (can-pay? state :runner
                                  (assoc eid :source card :source-type :ability)
                                  card nil :randomly-trash-from-hand 1)
-                     ["Card from grip" "Trash"]
-                     ["Trash"]))
+                       "Trash a random card from the grip")
+                     "Trash Mystic Maemi"])
      :player :runner
+     :msg (req (if (= target "Trash Mystic Maemi")
+                "to trash itself"
+                (msg "to " (decapitalize target))))
      :async true
-     :effect (req (if (= target "Trash")
-                    (do (system-msg state :runner "trashes Mystic Maemi")
-                        (trash state :runner eid card {:cause-card card}))
-                    (wait-for (pay state :runner
-                                   (make-eid state {:source card :source-type :ability})
-                                   card [:randomly-trash-from-hand 1])
-                              (system-msg state :runner (build-spend-msg (:msg async-result) "avoid trashing Mystic Maemi"))
-                              (effect-completed state side eid))))}
+     :effect (req (if (= target "Trash Mystic Maemi")
+                    (trash state :runner eid card {:cause-card card})
+                    (do (pay state :runner
+                             (make-eid state {:source card :source-type :ability})
+                             card [:randomly-trash-from-hand 1])
+                        (effect-completed state side eid))))}
```

Comment:
Pass the eid directly to pay. Then, remove the do and effect-completed.

Reply:
No need for `:source` and `:source-type`?

Reply:
If the ability requires that, then include them with `(assoc eid :source card :source-type :ability)`. I was writing this on my phone so was brief.